src/main/java/org/broadinstitute/consent/http/models/dataset_registration_v1/DatasetRegistrationSchemaV1UpdateValidator.java

```diff
@@ -62,11 +62,12 @@ public boolean validate(Study existingStudy, DatasetRegistrationSchemaV1 registr
         .filter(cg -> Objects.nonNull(cg.getConsentGroupName()))
         // If the dataset already has a name, this consent group is invalid
         .filter(cg -> {
-          Optional<Dataset> dataset = existingStudy
+          Optional<Dataset> dataset = Objects.nonNull(existingStudy.getDatasets()) ? existingStudy
               .getDatasets()
               .stream()
               .filter(d -> d.getDataSetId().equals(cg.getDatasetId()))
-              .findFirst();
+              .findFirst() :
+              Optional.empty();
```

Comment:
If you'd like, here is another way to handle this (we like `ListUtils.emptyIfNull` in TDR for cases like this):

```
import org.apache.commons.collections4.ListUtils;

…

Optional<Dataset> dataset =
  ListUtils.emptyIfNull(existingStudy.getDatasets())
    .stream()
    .filter(d -> d.getDataSetId().equals(cg.getDatasetId()))
    .findFirst();
```

Reply:
Thank you, that does look cleaner!